src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -237,10 +256,198 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
+          fixedBatch.push({
+            ...ele,
+            timestamp: earliestTimestamp,
+            blockNumber: earliestBlockNumber
+          })
+          continue
+        }
+        fixedBatch.push(ele)
+      }
+      return fixedBatch
+    }
+
+    const fixBlock11767088 = async (b: Batch): Promise<Batch> => {
+      // These just so happen to be a block number & timestamp that we are stuck at & so
+      // to get unstuck use these values.
+      const badBlockNumber = 11767088
+      const goodBlockNumber = 11767090
+      const fixedBatch: Batch = []
+      for (const ele of b) {
+        if (ele.blockNumber === badBlockNumber) {
+          this.log.warn('Fixing block 11767088!')
+          fixedBatch.push({
+            ...ele,
+            blockNumber: goodBlockNumber
+          })
+          continue
+        }
+        fixedBatch.push(ele)
+      }
+      return fixedBatch
+    }
+
+    if (this.autoFixBatchOptions.fixDoublePlayedDeposits) {
+      batch = await fixDoublePlayedDeposits(batch)
+    }
+    if (this.autoFixBatchOptions.fixDelayedTimestampAndBlockNumberHardcoded) {
+      batch = await fixDelayedTimestampAndBlockNumberHardcoded(batch)
+    }
+    if (this.autoFixBatchOptions.fixBlock11767088) {
+      batch = await fixBlock11767088(batch)
+    }
+    return batch
+  }
+
+  private async _fixQueueElement(queueIndex: number, queueElement: BatchElement): Promise<BatchElement> {
+    const [queueEleHash, timestamp, blockNumber] = await this.chainContract.getQueueElement(queueIndex)
+
+    if (timestamp > queueElement.timestamp && blockNumber > queueElement.blockNumber) {
+      this.log.warn('Double deposit detected!!! Fixing by skipping the deposit & replacing with a dummy tx.')
+      // This implies that we've double played a deposit.
+      // We can correct this by instead submitting a dummy sequencer tx
+      const wallet = Wallet.createRandom()
+      const gasLimit = 8_000_000
+      const gasPrice = 0
+      const chainId = 10
+      const nonce = 0
+      const rawTx = await wallet.signTransaction({
+        gasLimit,
+        gasPrice,
+        chainId,
+        nonce,
+        to: '0x1111111111111111111111111111111111111111',
+        data: '0x1234'
+      })
+      // tx: [0nonce, 1gasprice, 2startgas, 3to, 4value, 5data, 6v, 7r, 8s]
+      const tx = ethers.utils.RLP.decode(rawTx)
+      const dummyTx: EIP155TxData = {
+        sig: {
+          v: tx[6],
+          r: tx[7],
+          s: tx[8],
+        },
+        gasLimit,
+        gasPrice,
+        nonce,
+        target: tx[3],
+        data: tx[5]
+      }
+      // Sleep two miliseconds to ensure that the signature is unique.
+      // I am generating the signature baseed on the hash of the current time
+      // so I'm using time as a kind of global nonce & just to make sure that
+      // no two signatures are the same I wait for 2 miliseconds explicitly.
+      await new Promise((r) => setTimeout(r, 2))
```

Comment:
The signature is unique because the private key is ephemeral? I don't see how time plays a role in this

Reply:
This comment was leftover